featuregate/flag.go

```diff
@@ -10,11 +10,25 @@ import (
 	"go.uber.org/multierr"
 )
 
+// Flags related to feature gates.
+type Flags struct {
```

Comment:
Question, why just having Options in the `NewFlag` is not enough to extend capabilities?

Reply:
>  why just having Options in the NewFlag is not enough to extend capabilities?

How would you add the new flag from #7991 in this way? One would have to define the strict flag before defining the usual flag, I am not sure if that is even possible.

I am also not sure what the benefit of `Options` over this approach is. It is still [a breaking change](https://go.dev/play/p/LYJ7VL_kXwP).

Reply:
I am not arguing about breaking change, but about the fact that now this API is more complicated for normal users. Returning a struct that then I need decide which flag to install, etc.

If we want to allow use to construct multiple flags, then pass a FlagSet to the func and do that. Also still we should add Options to allow extensions in the future.

Reply:
I would do something like:

```golang
// NewFlags returns Flags that control building the global Registry.
func RegisterFlag(flag.FlagSet fs, reg *Registry, opts ...Option) Flags {
	...
        // One option can be to enable strict mode/to add a flag for that.
       // Though, the problem with flag is that it has to be defined first.
}
```

Reply:
PTAL at 1fb858c67, is that what you had in mind? This works for me